### ISSUE TITLE
Areas marked as buildings

### ISSUE THREAD

**Issue by jacopofar** [rustprooflabs/pgosm-flex#227]
## What version of PgOSM Flex are you using?

The current one from Docker hub, e4a19009b701

## What operating system, osm2pgsql, and PostgreSQL/PostGIS versions are you using?

The included PostGIS 12 + PostGIS 14, but the problem is probably in the Lua scripts

## What did you do exactly?

I imported the 2022-02-19 Geofabrik export of Berlin, Germany, then visualized it in QGIS.

## Issue description

After importing the file for the city of Berlin, I noticed that the building_polygon table contains a few polygons that are not bulding but rather city areas.

![grafik](https://user-images.githubusercontent.com/1280346/154845750-a1b6dc47-5f81-4d24-ae36-39b19622d7a1.png)

I'm not sure this is not the intended behavior, these are the not buildings but boundaries of postal codes. Not all the postal codes appear as buildings, only 3 of them:

https://www.openstreetmap.org/relation/1405771
https://www.openstreetmap.org/relation/1405640
https://www.openstreetmap.org/relation/1405638

they all have the attribute `boundary=postal_code` and `type=boundary`. The other postal codes end up in the `place_polygon` table, for example these ones:

https://www.openstreetmap.org/relation/1397900
https://www.openstreetmap.org/relation/1397881

The only common denominator I can see for the three postal codes above marked as buildings is that they have `addr:city=Berlin` while the ones marked as places do not.

**Comment by jacopofar** [comment]
Sorry, nevermind, I realized I was using a 4 months old Docker image. I have to adapt my scripts to the new one to check whether the issue is still there, there's no more run-all.lua`

**Comment by rustprooflabs** [comment]
I suspect this is still a problem in the latest too. What's the value of the `osm_type` for those records?  I'd guess it is `osm_type='address'`, if so it's being caught by logic added in https://github.com/rustprooflabs/pgosm-flex/issues/97. 
The decision to include/exclude address only "building" objects is handled here: https://github.com/rustprooflabs/pgosm-flex/blob/main/flex-config/style/building.lua#L54-L78

That includes the comment:

```
    -- Opting to include any addr: tag that was not excluded explicitly above
    --   This might be too wide of a net, but trying to be too picky risks
    --   excluding potentially important data
```

My initial hunch is the best approach would be to exclude the `boundary=postal_code` attribute, or even a wider net of exclusion using any `boundary` tag. The other option would to try to make the selection of `addr:` tags more specific, though I think this approach is trickier than excluding records we do not want in buildings.

**Comment by jacopofar** [comment]
After #228 it looks like this:

![grafik](https://user-images.githubusercontent.com/1280346/154861834-23e2cf48-cc74-4687-a103-5c66164abd1c.png)

**Comment by jacopofar** [comment]
I checked the database importing with both versions, and I don't see anything removed but the three post codes

![grafik](https://user-images.githubusercontent.com/1280346/154862113-40ce72fb-23b2-41e8-b0a0-b3c5e1524c35.png)

**Comment by rustprooflabs** [comment]
Fixed in #228 and #230.  Merged into `dev` branch and available on Docker Hub `docker pull rustprooflabs/pgosm-flex:dev` hash `046bfa49decf`. Will merge into `main` branch and `latest` image soon.

**Comment by rustprooflabs** [comment]
This is merged into `main` branch, pushed updated `:latest` Docker image.
